beetmoverscript/docker.d/worker.yml

```diff
@@ -30,6 +30,8 @@ bucket_config:
     $match:
       'COT_PRODUCT == "firefox" && (ENV == "dev" || ENV == "fake-prod")':
         dep:
+          gcs_credentials: { "$eval": "GCS_CREDENTIALS" }
+          fail_task_on_gcs_error: False
```

Comment:
I'm not sure I love this layout.

a) it assumes that S3 is the main way of doing things, and GCS is a secondary way of doing things, and those assumptions are baked into the code.
b) when we're ready to cut over so that GCS is the primary way of doing things, we may need to land a very large code change. (I'm guessing -- are the buckets named the same in GCS as S3? If so, it's a smaller delta, but still.)

I'd probably look something like

```
clouds:
    s3:
       credentials: ...
       enabled: True
       fail_on_error: True
       ... # if url_prefix verification etc are cloud-specific, list them here rather than in the shared config space
    gcs:
       ...
```

This takes more up-front work, but if we want to require both clouds upload successfully, that's a single boolean flip. If we are ready to cut over to GCS and stop uploading to S3, that's another boolean flip.

This may be a me thing: I prefer to make things more configuration-driven than hardcoded code-logic driven.

Reply:
Heh that's how I started with my changes. But after realizing how tightly coupled with `bucket_config` it is, I decided to move the GCS config in with s3 stuff.

It's fairly easy to move it out, but it will still require more work to be able to cutoff S3.
Should I add the ability for us to cutoff already, or should we just support adding GCS for now and cutoff later?

Reply:
This probably depends on how urgent the request is. If we have time, then doing the work ahead of time makes sense to me. If getting GCS populated sooner is urgent, we might have to land this in phases.

Reply:
I'd guess we have a month or two before this becomes super urgent.